Polars `pivot_longer` implementation

samukweku opened the pull request:
# PR Description

Please describe the changes proposed in the pull request:

- `pivot_longer` implementation for polars
-  `pivot_longer_spec` for more granular control


**This PR is related to #1352 .**

<!-- As you go down the PR template, please feel free to delete sections that are irrelevant. -->

# PR Checklist

<!-- This checklist exists for newcomers who are not yet familiar with our requirements. If you are experienced with
the project, please feel free to delete this section. -->

Please ensure that you have done the following:

1. [x] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.
<!-- Doing this helps us keep the commit history much cleaner than it would otherwise be. -->
2. [x] If you're not on the contributors list, add yourself to `AUTHORS.md`.
<!-- We'd like to acknowledge your contributions! -->
3. [x] Add a line to `CHANGELOG.md` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

# Automatic checks

There will be automatic checks run on the PR. These include:

- Building a preview of the docs on Netlify
- Automatically linting the code
- Making sure the code is documented
- Making sure that all tests are passed
- Making sure that code coverage doesn't go down.

# Relevant Reviewers

<!-- Finally, please tag relevant maintainers to review. -->

Please tag maintainers to review.

- @ericmjl

Commit:
add make_clean_names function that can be applied to polars

Commit:
add make_clean_names function that can be applied to polars

Commit:
add examples for make_clean_names

Commit:
add examples for make_clean_names

Commit:
changelog

Commit:
changelog

Commit:
limit import location for polars

Commit:
limit import location for polars

Commit:
limit import location for polars

Commit:
fix polars in environment-dev.yml

Commit:
install polars in doctest

Commit:
limit polars imports - user should have polars already installed

Commit:
limit polars imports - user should have polars already installed

Commit:
use subprocess.run

Commit:
add subprocess.devnull to docstrings

Commit:
add subprocess.devnull to docstrings

Commit:
add subprocess.devnull to docstrings

Commit:
add subprocess.devnull to docstrings

Commit:
add subprocess.devnull to docstrings

Commit:
add os.devnull

Commit:
add polars as requirement for docs

Commit:
add polars to tests requirements

Commit:
add polars to tests requirements

Commit:
delete irrelevant folder

Commit:
changelog

Commit:
changelog

Commit:
create submodule for polars

Commit:
create submodule for polars

Commit:
fix doctests

Commit:
fix doctests

Commit:
fix tests; add polars to documentation

Commit:
fix tests; add polars to documentation

Commit:
fix tests; add polars to documentation

Commit:
import janitor.polars

Commit:
control docs output for polars submodule

Commit:
control docs output for polars submodule

Commit:
exclude functions in docs rendering

Commit:
exclude functions in docs rendering

Commit:
show_submodules=true

Commit:
fix docstring rendering for polars

Commit:
fix docstring rendering for polars

Commit:
Expression -> expression

Commit:
Expression -> expression

ericmjl committed:
Merge dev into samukweku/polars_clean_names

Commit:
rename functions.py

Commit:
rename functions.py

Commit:
pivot_longer implemented for polars

Commit:
pivot_longer implemented for polars

Commit:
changelog

Commit:
changelog

Commit:
keep changes related only to pivot_longer

Commit:
keep changes related only to pivot_longer

ericmjl commented:
<!-- NETLIFY DEPLOY COMMENT GENERATED BY ACTIONS_NETLIFY -->
🚀 Deployed on https://deploy-preview-1355--pyjanitor.netlify.app

Commit:
pd -> pl

Commit:
pd -> pl

Commit:
df.pivot_longer -> df.janitor.pivot_longer

Commit:
df.pivot_longer -> df.janitor.pivot_longer

Commit:
pd -> pl

Commit:
pd -> pl

Commit:
add >>> df

Commit:
add >>> df

Commit:
keep changes related only to polars pivot_longer

Commit:
code update

Commit:
names_transform if there is not dot value

Commit:
add support for lazyframe

Commit:
add support for lazyframe

Commit:
fix column order in docs

Commit:
fix column order in docs

Commit:
fix column order in docs

Commit:
fix column order in docs

Commit:
separate namespaces for lazyframe and eager dataframe

Commit:
separate namespaces for lazyframe and eager dataframe

Commit:
separate namespaces for lazyframe and eager dataframe

Commit:
improve perf if no dot_value and values_to is a str

Commit:
fix docs

Commit:
handle deprecation warning for pd.unique

ericmjl committed:
Merge dev into samukweku/polars_pivot_longer

samukweku committed:
Merge branch 'dev' into samukweku/polars_pivot_longer

Commit:
added pivot_longer_spec

Commit:
added pivot_longer_spec

Commit:
update docs

Commit:
update docs

Commit:
update docs

Commit:
update docs

Commit:
update docs

Commit:
update docs

Commit:
update docs

Commit:
update docs

Commit:
update docs

Commit:
reduce repetitiveness in function calls

Commit:
reduce repetitiveness in function calls

Commit:
update docs

Commit:
update docs

Commit:
update docs

Commit:
update docs

Commit:
refactor code - create spec then unpivot

Commit:
refactor code - create spec then unpivot

Commit:
update docs

Commit:
update docs

Commit:
update docs

Commit:
update docs

Commit:
add check for spec dataframe

Commit:
add check for spec dataframe

Commit:
fix doc fail

Commit:
fix doc fail

Commit:
exclude non polars code

ericmjl committed:
Merge dev into samukweku/polars_pivot_longer

ericmjl committed:
Merge dev into samukweku/polars_pivot_longer

Commit:
change column selection

ericmjl committed:
Merge dev into samukweku/polars_pivot_longer

Commit:
call join only if necessary

Commit:
rechunk

Commit:
names_transform should be a pl.Expr

Commit:
names_transform should be a pl.Expr

Commit:
adjust tests

Commit:
__future__

Commit:
__future__

Commit:
changelog

Commit:
changelog

Commit:
compute only when necessary

Commit:
improve performance

Commit:
fix docs

Commit:
fix docs

samukweku committed:
non-equi join rewrite numba (#1341)

- reimplementation of the algorithm that powers the non-equi join in numba
- more info in the docs regarding range joins, and where numba can be helpful
- improved performance for numba for range joins and multiple non equi joins when getting first/last match

samukweku committed:
non-equi join rewrite numba (#1341)

- reimplementation of the algorithm that powers the non-equi join in numba
- more info in the docs regarding range joins, and where numba can be helpful
- improved performance for numba for range joins and multiple non equi joins when getting first/last match

samukweku committed:
Implement clean_names for polars (#1351)

As per title. Tests included. h/t Samuel Oranyeli for making this happen!

ericmjl commented:
@samukweku looks like we are getting merge conflicts on this PR. I'm a bit hesitant to edit the files as I'm not 100% sure what the right final state should look like. Could you resolve the conflicts please?

ericmjl reviewed (changes requested):
Marking this with a "Request changes" to remind myself that we have merge conflicts that I'm not confident in resolving and that I would need @samukweku's input to resolve them.

samukweku committed:
Add polars support for `janitor.io.xlsx_table` (#1357)

* add make_clean_names function that can be applied to polars

* add examples for make_clean_names

* changelog

* limit import location for polars

* limit import location for polars

* fix polars in environment-dev.yml

* install polars in doctest

* limit polars imports - user should have polars already installed

* use subprocess.run

* add subprocess.devnull to docstrings

* add subprocess.devnull to docstrings

* add subprocess.devnull to docstrings

* add subprocess.devnull to docstrings

* add os.devnull

* add polars as requirement for docs

* add polars to tests requirements

* delete irrelevant folder

* changelog

* create submodule for polars

* fix doctests

* fix tests; add polars to documentation

* fix tests; add polars to documentation

* import janitor.polars

* control docs output for polars submodule

* exclude functions in docs rendering

* exclude functions in docs rendering

* show_submodules=true

* fix docstring rendering for polars

* Expression -> expression

* rename functions.py

* pivot_longer implemented for polars

* changelog

* keep changes related only to pivot_longer

* pd -> pl

* pd -> pl

* df.pivot_longer -> df.janitor.pivot_longer

* df.pivot_longer -> df.janitor.pivot_longer

* pd -> pl

* pd -> pl

* add >>> df

* add >>> df

* keep changes related only to polars pivot_longer

* add polars support to read_commandline

* remove irrelevant files

* minor edit to docs

* xlsx_table now supports polars

---------

Co-authored-by: samuel.oranyeli <samuel.oranyeli@grow.inc>
Co-authored-by: Eric Ma <ericmjl@users.noreply.github.com>

samukweku committed:
Add support for polars in `xlsx_cells` (#1358)

Commit:
updates

Commit:
fix conflicts

samukweku committed:
Merge branch 'dev' into samukweku/polars_pivot_longer

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Commit:
fix conflicts

Commit:
fix changelog

ericmjl committed:
Merge dev into samukweku/polars_pivot_longer

ericmjl commented:
LGTM! Thank you, @samukweku, for the hard work you've been putting in here!
